cmd/flowserver/middleware.go

```diff
@@ -44,23 +40,3 @@ func requestLogger() func(next http.Handler) http.Handler {
 		return http.HandlerFunc(fn)
 	}
 }
-
-func traceErrors() func(next http.Handler) http.Handler {
-	return func(next http.Handler) http.Handler {
-		fn := func(w http.ResponseWriter, r *http.Request) {
-			body := bytes.Buffer{}
-			r.Body = ioutil.NopCloser(io.TeeReader(r.Body, &body))
-			ww := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
-			next.ServeHTTP(ww, r)
-
-			// we are returning a system error of some kind, log the incoming request body
-			if ww.Status() >= 500 && strings.ToLower(r.Method) == "post" {
```

Comment:
so this never happens - panics end up being converted to 502s at the nginx level

Reply:
are we still logging the request somewhere? Seems `Recoverer` actually traps the panic then changes the response to a 500 so maybe we were just missing that piece?

Reply:
Maybe - but I've now changed it to use a combination of raven's own recovery middleware (not made for chi) and chi's own implementation. The recovery middleware comes before the request logger middleware - so the former recovers from the panic, logs it to sentry along with the request, converts it to 500 response - the latter will then log that. In theory. Not the easiest thing to test.